### PR TITLE
JS client improvements

### LINKED ISSUE #10: sinopia server testing
- [x] investigate ways to test that sinopia server (i.e. trellis-ext-db and our augmentations to it) correctly implements the LDP operations we care about.
- [x] start writing some tests

### PR DESCRIPTION
**Pull request by jmartin-sul**
* update the swagger spec to use `/repository` as the root of our paths (instead of `/resources`), as per pairing with @cmh2166 and @jermnelson a few weeks ago to explore trellis functionality.
  * that change is not backwards compatible, so major version bump to the spec
* start implementing some of the test stubs that `swagger-codegen` provided us.
* switch the generated client code to use [promises](https://www.promisejs.org/) instead of [superagent](https://github.com/visionmedia/superagent) callbacks.  adjust tests accordingly.

_note_:  tests currently hardcoded to expect trellis to be running on `http://localhost:8080` (`:8081` for `/healthcheck`).  ~~we'll prob need to fix that to play nicely with CI (_edit_: and indeed, this is breaking the build).~~

closes #10

### CODE REVIEW

**Comment by cmh2166** [comment]
lgtm. testing the client code now with trellis running locally. wondering what to do about the failing coverage test.

**Comment by cmh2166** [comment]
> @cmh2166 said:
> 
> > wondering what to do about the failing coverage test
> 
> i think i need to work with @atz to fix the tests so that they can take a `<hostname>:<port>` that works for circle. i was already figuring that the hardcoded `localhost` was a hack that'd have to go.

yeah, that makes sense, i saw your comment on hardcoding above. have you and @atz gotten further on the orchestration questions around where to deploy trellis (and possibly have the swagger spec there, with the client code eventually going into front end codebases)/

**Comment by jmartin-sul** [comment]
we've discussed it some, but i don't think we came up with any clear answers yet.  i just pinged him on slack to try to pair for a bit this afternoon on it.

**Comment by acoburn** [comment]
FYI with this commit (https://github.com/trellis-ldp/trellis-ext-db/commit/497144deaa09ab43aa90f78f92fd8e4eb1ccaef9) you can now set the Trellis HTTP port with an environment variable such as: `TRELLIS_HTTP_PORT=9999` and/or `TRELLIS_ADMIN_HTTP_PORT=9998`

**Comment by jmartin-sul** [comment]
ok, i cleaned up the commits to get a passing CI build using the nascent swagger-codegen based tests.

left to do:
[ ] re-gen JS client and tests based on the changes @cmharlow made to the swagger spec in this PR.  #30
[ ] implement the tests.  #29

File: .circleci/config.yml [hunk 1]
```diff
@@ -1,6 +1,5 @@
 defaults: &defaults
```
Comment (jermnelson) [inline .circleci/config.yml]: Like how you're able to get getting Trellis working with circle-ci and Docker

File: sinopia_client/README.md [hunk 2]
```diff
@@ -125,19 +122,19 @@ All URIs are relative to *https://localhost*
 
 Class | Method | HTTP request | Description
```
Comment (jermnelson) [inline sinopia_client/README.md]: General question, can the documentation be generated without re-generating the code base?

**Comment by jermnelson** [comment]
Hi John, looks good to me. Made some minor comments but nothing blocking. Do you think this client code should be an imported node module by the Sinopia Editor for making API calls? 

Thanks,
Jeremy

File: sinopia_client/README.md (hunk 2 above)
Comment (jmartin-sul) [reply sinopia_client/README.md]: assuming i'm understanding this correctly, i don't think that's possible, at least with the `swagger-codegen` tool used here.  i'm pretty sure the documentation is generated from the swagger spec and not the JS code, so i don't think you'd be able to generate these docs from arbitrary code, if that's what you mean.

maybe what's desired in this case, for the non-swagger-specified JS code, is something like YARD/rubydoc/javadoc?  is there a JS equivalent for that?  that'd require keeping method level comments up to date, but that at least lives with the code, and so seems less onerous to update and generate off of than hand-writing docs.

**Comment by jmartin-sul** [comment]
@jermnelson or @ndushay, might i get one of you to merge if there are no objections?  or to at least approve so this doesn't look like a totally unwarranted self-merge? 😁

**Pull request merged**